Marlin/src/HAL/HAL_STM32F1/fastio_STM32F1.h

```diff
@@ -30,7 +30,7 @@
 #include <libmaple/gpio.h>
 
 #define READ(IO)              (PIN_MAP[IO].gpio_device->regs->IDR & (1U << PIN_MAP[IO].gpio_bit) ? HIGH : LOW)
-#define WRITE(IO,V)           (PIN_MAP[IO].gpio_device->regs->BSRR = (1U << PIN_MAP[IO].gpio_bit) << (16 * !(bool)V))
+#define WRITE(IO,V)           (PIN_MAP[IO].gpio_device->regs->BSRR = (1U << PIN_MAP[IO].gpio_bit) << (16 * !((bool)V)))
```

Comment:
Wow, that is subtle as f***. Amazing find that I'd not have found in 1m years.

Reply:
Took me almost as long to troubleshoot it using the only way I know how... Serial Echo and changing the value of the LED pin, haha. 🤣 

Turns out in my case that the results were even more exceptional - the statement was evaluating to a integer that corresponded to the register of a different pin!! turning on/off the extruder heater was turning on the bed heater! I felt I was going insane at times! 🤕

Reply:
The icon was just lucky! 🤷‍♂️  😃